src/main/java/com/sap/sgs/phosphor/fosstars/model/qa/AbstractVerifier.java

```diff
@@ -62,4 +64,36 @@ public void verify() throws VerificationFailedException {
       }
     }
   }
+
+  /**
+   * Verify a score value against a test vector.
+   *
+   * @param vector The test vector.
+   * @param scoreValue The score value to be verified.
+   * @param index An index of the test vector.
+   * @return A result of the verification.
+   */
+  protected static TestVectorResult testResultFor(
+      TestVector vector, ScoreValue scoreValue, int index) {
+
+    if (vector.notApplicableScore() && scoreValue.isNotApplicable()) {
+      return new TestVectorResult(vector, index++, scoreValue,
+          Status.PASSED, "Ok, score is N/A as expected");
+    }
+
+    if (vector.notApplicableScore() && !scoreValue.isNotApplicable()) {
```

Comment:
What about the case where vector is applicable and scoreValue is not applicable?

Reply:
I thought that `containsExpected()` handles that but in fact it doesn't. I'll fix it.

Reply:
That's a very good catch @Sachpat . Now it should be fixed. I also improved a test for the `ScoreVerifier`.